### ISSUE TITLE
Actioncable Allowed Request Origins Ignores -p flag in 5.0.0.rc1

### ISSUE THREAD

**Issue by michaeldever** [rails/rails#25406]
### Steps to reproduce

Create a new rails project with an actioncable channel.

```bash
rails new example
rails g channel example
```

Setup channel coffeescript as follows:

```coffeescript
App.example = App.cable.subscriptions.create "ExampleChannel",
  connected: ->
    console.log('connected');

  disconnected: ->
    console.log('disconnected');

  received: (data) ->
    console.log('received');
```

Set up channel object as follows:

```ruby
class ExampleChannel < ApplicationCable::Channel
  def subscribed
    stream_from 'examples'
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end
end
```

Create an example controller:

```bash
rails g controller example index
```

Starting the server with `rails s` and navigating to `http://localhost:3000/example` results in `connected` being printed in the javascript console.

Starting the server with `rails s -p 5000` and navigating to `http://localhost:5000/example` results in the following being printed to the console:

`
WebSocket connection to 'ws://localhost:5000/cable' failed: Error during WebSocket handshake: Unexpected response code: 404
`

### Expected behavior
As with `rails s` I would expect `rails s -p 5000` to result in `connected` to be displayed in the javascript console, as nothing else changes in the app in between.

### Actual behavior
`WebSocket connection to 'ws://localhost:5000/cable' failed: Error during WebSocket handshake: Unexpected response code: 404` being printed in the javascript console.

`Request origin not allowed: http://localhost:5000
Failed to upgrade to WebSocket (REQUEST_METHOD: GET, HTTP_CONNECTION: Upgrade, HTTP_UPGRADE: websocket)` printed in the server log.

### System configuration
**Rails version**: 5.0.0.rc1

**Ruby version**: 2.3.0

#### Resolving The Problem
By setting the set of allowed request origins, as follows, from environmental variables at initialization in, say, `config/initializers/action_cable.rb` and starting the server with `PORT=5000 rails s`, I can eliminate this problem, but would expect that changing the server port would not result in a change in behaviour.

```ruby
Rails.application.configure do
  host = ENV.fetch('HOST', 'localhost')
  port = ENV.fetch('PORT', '3000')

  config.action_cable.allowed_request_origins = ["http://#{host}:#{port}", "https://#{host}:#{port}"]
end
```

**Comment by maclover7** [comment]
This is expected behavior -- `localhost:3000` is set as `config.action_cable.allowed_request_origins` in development mode --> https://github.com/rails/rails/blob/master/actioncable/lib/action_cable/engine.rb#L25

**Comment by prathamesh-sonpatki** [comment]
We can set https://github.com/rails/rails/blob/61a7239e405982e78de15bd86583514248e6f525/actioncable/lib/action_cable/engine.rb#L25 in the generated `development.rb` instead of action cable railtie and add a comment for users so that they can change it as per their needs. 

production.rb already has such comment - https://github.com/rails/rails/blob/61a7239e405982e78de15bd86583514248e6f525/railties/lib/rails/generators/rails/app/templates/config/environments/production.rb.tt#L43.

Though this will not solve the problem entirely, it will make it a bit more discoverable.

**Comment by jeremy** [comment]
Thanks @michaeldever. Could you open a PR with your suggested resolution?

**Comment by maclover7** [comment]
👍 for changing this behavior

**Comment by michaeldever** [comment]
Will do, won't get a chance until tomorrow evening after work. What's the best way to resolve this... An actioncable initializer like I have used to resolve already, based on ENV variables or an addition to the existing development.rb file?

**Comment by michaeldever** [comment]
`rails s` sets the PORT environment variable if -p is set, so both approaches should work just fine using this.

**Comment by michaeldever** [comment]
I had a look at this, and it seems there is no easy/tidy way to get the port available at the point that `development.rb` is loaded, without resorting to something like [this](http://stackoverflow.com/a/13839413), which doesn't seem like the right way to do it to me. The same goes for setting it in the [action_cable.set_configs initializer](https://github.com/rails/rails/blob/61a7239e405982e78de15bd86583514248e6f525/actioncable/lib/action_cable/engine.rb#L25).

Can anyone suggest a better way of doing this? @jeremy @prathamesh-sonpatki @maclover7

**Comment by jeremy** [comment]
Alternatively, could add support for wildcard port matches and use that in dev.

**Comment by michaeldever** [comment]
@jeremy Hah, I was just thinking that too, but I wanted to err on the side of being precise about what's allowed. That may be the best solution for now alright, I'll submit a PR in a little while.

**Comment by maclover7** [comment]
Fixed by #25420.
